### PR TITLE
Add option to extend the search settings component

### PR DESCRIPTION
**Pull request by pvargas**
Adds an optional param to the QuerySettings component.
This change doesn't visually affect the UI since it is to be used by downstream projects.

### CODE REVIEW

**Comment by jlcsmith** [comment]
build now

**Comment by maiiiiiiii** [comment]
Screenshot 2024-02-08 at 6 08 07 PM

**Review by maiiiiiiii — APPROVED** [review]

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/query-settings/query-settings.tsx [hunk 1]
```diff
@@ -27,17 +27,19 @@ import { useConfiguration } from '../../js/model/Startup/configuration.hooks'
 
 type Props = {
   model: QueryType
+  Extensions?: any
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/query-settings/query-settings.tsx:30]: Is there a type that could be used for `Extensions`?

**Review by sgalla — APPROVED** [review]

**Comment by pvargas** [comment]
build now

**Comment by jlcsmith** [comment]
build now

**Comment by jlcsmith** [comment]
build now

**Pull request merged**